reactos/drivers/usb/usbohci_new/usbohci.h

```diff
@@ -0,0 +1,253 @@
+#ifndef USBOHCI_H__
+#define USBOHCI_H__
+
+#include <ntddk.h>
+#include <windef.h>
+#include <stdio.h>
+#include <hubbusif.h>
+#include <usbbusif.h>
+#include <usbdlib.h>
+#include <drivers/usbport/usbmport.h>
+
+#include "hardware.h"
+
+extern USBPORT_REGISTRATION_PACKET RegPacket;
+
+#define OHCI_HCD_ED_FLAG_CONTROLL      0x00000001
+#define OHCI_HCD_ED_FLAG_RESET_ON_HALT 0x00000008
+#define OHCI_HCD_ED_FLAG_NOT_ACCESSED  0x00000010
+
+#define OHCI_HCD_TD_FLAG_ALLOCATED       0x00000001
+#define OHCI_HCD_TD_FLAG_PROCESSED       0x00000002
+#define OHCI_HCD_TD_FLAG_CONTROLL_STATUS 0x00000004
+#define OHCI_HCD_TD_FLAG_DONE            0x00000008
+#define OHCI_HCD_TD_FLAG_NOT_ACCESSED    0x00000010
+
+#define OHCI_MAX_ISO_TRANSFER_SIZE        0x10000
+#define OHCI_MAX_CONTROL_TRANSFER_SIZE    0x10000
+#define OHCI_MAX_BULK_TRANSFER_SIZE       0x40000
+#define OHCI_MAX_INTERRUPT_TRANSFER_SIZE  0x1000
+
+#define OHCI_MAX_ISO_TD_COUNT        64
+#define OHCI_MAX_CONTROL_TD_COUNT    38
+#define OHCI_MAX_BULK_TD_COUNT       68
+#define OHCI_MAX_INTERRUPT_TD_COUNT  4
+
+#define OHCI_TRANSFER_FLAGS_SHORT_TRANSFER_OK  1
+
+typedef struct _OHCI_TRANSFER *POHCI_TRANSFER;
+
+typedef union _OHCI_HW_TRANSFER_DESCRIPTOR {
+  struct {
+    OHCI_TRANSFER_DESCRIPTOR gTD; // must be aligned to a 16-byte boundary
+    USB_DEFAULT_PIPE_SETUP_PACKET SetupPacket;
+    ULONG Padded[2];
+  };
+  struct {
+    OHCI_ISO_TRANSFER_DESCRIPTOR iTD; // must be aligned to a 32-byte boundary
+  };
+} OHCI_HW_TRANSFER_DESCRIPTOR, *POHCI_HW_TRANSFER_DESCRIPTOR;
+
+C_ASSERT(sizeof(OHCI_HW_TRANSFER_DESCRIPTOR) == 32);
```

Comment:
If the entire struct is 32 bytes, the second member structure will not be aligned on 32-byte boundary.

Reply:
Why?